MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -8084,5 +8091,117 @@ private void gettime()
+            var rawText = Encoding.ASCII.GetString(Module.Memory.GetString(textPointer, true));
+            var split = rawText.Split('\r');
+            var lineNumber = 0;
+            for (var i = 0; i < split.Length; i++)
+            {
+                if (split[i].Length < 79)
+                {
+                    answersString.Append($"LINE{i}={split[i]}\0");
+                    lineNumber++;
+                    continue;
+                }
+
+                //Too much data for one line, we need to "word wrap."
+                while (split[i].Length > 79)
+                {
+                    answersString.Append($" LINE{lineNumber}={split[i].Substring(0, 79)}\0");
+                    split[i] = split[i][79..];
+                    lineNumber++;
+                }
+            }
+
+            //Double Null Terminated
+            answersString.Append('\0');
+
+            //FSDROOM
+            var resourceManager = new ResourceManager();
+
+            var template = resourceManager.GetResource("MBBSEmu.Assets.fseTemplate.ans");
+            var fieldSpec = resourceManager.GetResource("MBBSEmu.Assets.fseFieldSpec.txt");
+
+            var fsdBufferPointer = Module.Memory.GetOrAllocateVariablePointer($"FSD-TemplateBuffer-{ChannelNumber}", 0x2000);
+            var fsdFieldSpecPointer = Module.Memory.GetOrAllocateVariablePointer($"FSD-FieldSpec-{ChannelNumber}", 0x2000);
+
+            //Zero out FSD Memory Areas
+            Module.Memory.SetZero(fsdBufferPointer, 0x2000);
+            Module.Memory.SetZero(fsdFieldSpecPointer, 0x2000);
+
+            //Hydrate FSD Memory Areas with Values
+            Module.Memory.SetArray(fsdBufferPointer, template);
+            Module.Memory.SetArray(fsdFieldSpecPointer, fieldSpec);
+
+            //Establish a new FSD Status Struct for this Channel
+            var channelFsdscb = Module.Memory.GetOrAllocateVariablePointer($"FSD-Fsdscb-{ChannelNumber}", FsdscbStruct.Size);
+            var newansPointer = Module.Memory.GetOrAllocateVariablePointer($"FSD-Fsdscb-{ChannelNumber}-newans", 0x400);
+
+            //Declare flddat -- allocating enough room for up to 100 fields
+            var fsdfldPointer = Module.Memory.GetOrAllocateVariablePointer($"FSD-Fsdscb-{ChannelNumber}-flddat", FsdfldStruct.Size * 100);
+
+            var fsdStatus = new FsdscbStruct(Module.Memory.GetArray(channelFsdscb, FsdscbStruct.Size))
+            {
+                fldspc = fsdFieldSpecPointer,
+                flddat = fsdfldPointer,
+                newans = newansPointer
+            };
+
+            Module.Memory.SetArray($"FSD-Fsdscb-{ChannelNumber}", fsdStatus.Data);
+
+            //FSDAPR
+            var fsdAnswersPointer = Module.Memory.GetOrAllocateVariablePointer($"FSD-Answers-{ChannelNumber}", 0x800);
+            Module.Memory.SetArray($"FSD-Answers-{ChannelNumber}", Encoding.ASCII.GetBytes(answersString.ToString()));
```

Comment:
+ null terminator?

Reply:
`answerString` is compiled as double null terminated prior